### PR TITLE
audit: 3.1.1 -> 3.1.2

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

https://github.com/linux-audit/audit-userspace/releases/tag/v3.1.2

Note: all patches have been fixed in upstream and are no longer required.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/audit/default.nix [hunk 1]
```diff
@@ -18,20 +18,14 @@
...
+    hash = "sha256-wLF5LR8KiMbxgocQUJy7mHBZ/GhxLJdmnKkOrhA9KH0=";
   };
 
-  patches = [
-    ./000-fix-static-attribute-malloc.diff
-    ./001-ignore-flexible-array.patch
-  ];
-
   postPatch = ''
-    sed -i 's,#include <sys/poll.h>,#include <poll.h>\n#include <limits.h>,' audisp/audispd.c
```
Comment (soupglasses) [inline pkgs/os-specific/linux/audit/default.nix:34]: What is the reasoning for removing this sed? I cannot find what changed to allow its removal.

File: pkgs/os-specific/linux/audit/default.nix [hunk 2]
```diff
@@ -18,20 +18,14 @@
...
+  version = "3.1.2";
 
   src = fetchurl {
     url = "https://people.redhat.com/sgrubb/audit/audit-${finalAttrs.version}.tar.gz";
-    hash = "sha256-RuRrN2I8zgnm7hNOeNZor8NPThyHDIU+8S5BkweM/oc=";
+    hash = "sha256-wLF5LR8KiMbxgocQUJy7mHBZ/GhxLJdmnKkOrhA9KH0=";
   };
 
-  patches = [
-    ./000-fix-static-attribute-malloc.diff
```
Comment (soupglasses) [inline pkgs/os-specific/linux/audit/default.nix:29]: Merged with `v3.1.2`: https://github.com/linux-audit/audit-userspace/commit/e8c683b3ecef2d2d0908c5b8b22499c3206350b0

File: pkgs/os-specific/linux/audit/default.nix [hunk 3]
```diff
@@ -18,20 +18,14 @@
...
-    ./001-ignore-flexible-array.patch
```
Comment (soupglasses) [inline pkgs/os-specific/linux/audit/default.nix:30]: Merged with `v3.1.2`: https://github.com/linux-audit/audit-userspace/commit/1ca7e2b07bdc962695611ee1d0852f549d5ca21c

File: pkgs/os-specific/linux/audit/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/os-specific/linux/audit/default.nix:34]: Hello, thanks for the review!

This regex does not match anything since v3 with this commit (from nov. 2018):

https://github.com/linux-audit/audit-userspace/commit/6d6c65e8e374ce31037e20b1cdc314808efd0e3c#diff-035369519d0ce9cf625d86e351f8a1f5d8dc0609349b0775f30d82b30418a0d5L34

Comment (soupglasses) [reply pkgs/os-specific/linux/audit/default.nix:34]: Ah i see. I am curious for why it was changed, as there does exist two other cases of `<sys/poll.h>` in other files, but are both 15 year old and not recent.
- https://github.com/linux-audit/audit-userspace/blob/d1ccd8f642e11ad65cdc8c7091e8ef0304a3fd3d/lib/netlink.c#L30
- https://github.com/linux-audit/audit-userspace/blob/d1ccd8f642e11ad65cdc8c7091e8ef0304a3fd3d/lib/netlink.c#L30

Comment (soupglasses) [reply pkgs/os-specific/linux/audit/default.nix:34]: From a git blame, i can see that the line was added for presumably musl support. https://github.com/NixOS/nixpkgs/commit/32b472dd9d69f7cb5fff32058767c72d0d9799e3

File: pkgs/os-specific/linux/audit/default.nix [hunk 4]
```diff
@@ -68,12 +62,12 @@ stdenv.mkDerivation (finalAttrs: {
...
+  meta = with lib; {
     homepage = "https://people.redhat.com/sgrubb/audit/";
     description = "Audit Library";
     changelog = "https://github.com/linux-audit/audit-userspace/releases/tag/v${finalAttrs.version}";
-    license = lib.licenses.gpl2Plus;
-    maintainers = with lib.maintainers; [ AndersonTorres ];
-    platforms = lib.platforms.linux;
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ AndersonTorres ];
+    platforms = platforms.linux;
```
Comment (soupglasses) [inline pkgs/os-specific/linux/audit/default.nix:71]: Minor, but this should either be forked out into a separate commit, or not done at all.

**Review by soupglasses — CHANGES_REQUESTED** [review]
Overall this is looking really good, only really this single nitpick which stands out to me.

File: pkgs/os-specific/linux/audit/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/os-specific/linux/audit/default.nix:71]: It does not break the semantics, then there is no difference in doing this in a separate commit.

But let's not pursue this bikeshedding.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/os-specific/linux/audit/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/os-specific/linux/audit/default.nix:71]: This is my personal preference, based on the fact I am not fond of `with` - as explained many times, it is an anti-pattern:

https://nix.dev/recipes/best-practices#with-scopes

File: pkgs/os-specific/linux/audit/default.nix [hunk 5]
```diff
@@ -68,12 +62,12 @@ stdenv.mkDerivation (finalAttrs: {
 
   enableParallelBuilding = true;
 
-  meta = {
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/audit/default.nix:65]: Undo this factoring

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/audit/default.nix (hunk 4 above)
Comment (soupglasses) [reply pkgs/os-specific/linux/audit/default.nix:71]: Yeah, its very much an anti-pattern. But inside of places like `meta` its one of the few places i find it okay.

Ultimately, this refactor should either be done in another commit, or not at all, since the `meta` does not do any damage being one way or the other.

File: pkgs/os-specific/linux/audit/default.nix (hunk 5 above)
Comment (anthonyroussel) [reply pkgs/os-specific/linux/audit/default.nix]: Done, thanks for the review
